libpod/container_internal_linux.go

```diff
@@ -2273,7 +2296,12 @@ func (c *Container) updateHosts(path string) error {
 	if err != nil {
 		return err
 	}
-	c.state.BindMounts["/etc/hosts"] = newHosts
+
+	err = c.mountIntoRootDirs("/etc/hosts", newHosts)
```

Comment:
Something like
```suggestion
	if err := c.mountIntoRootDirs("/etc/hosts", newHosts); err != nil {
```
is what I think @rhatdan was looking for

Reply:
Thanks, I did it.